financial-connections/src/main/java/com/stripe/android/financialconnections/features/partnerauth/PartnerAuthViewModel.kt

```diff
@@ -262,25 +265,19 @@ internal class PartnerAuthViewModel @Inject constructor(
                 )
             )
             requireNotNull(authSession)
-            val result = cancelAuthorizationSession(authSession.id)
-            if (authSession.isOAuth) {
-                // For OAuth institutions, create a new session and navigate to its nextPane (prepane).
-                logger.debug("Creating a new session for this OAuth institution")
-                // Send retry event as we're presenting the prepane again.
-                postAuthSessionEvent(authSession.id, AuthSessionEvent.Retry(Date()))
-                // for OAuth institutions, we remain on the pre-pane,
-                // but create a brand new auth session
-                setState { copy(authenticationStatus = Uninitialized) }
-                createAuthSession()
+            if (url.isNullOrEmpty()) {
+                // client did not receive any deeplink -> retrieve the auth session and check its status
+                val retrievedAuthSession = retrieveAuthorizationSession(authSession.id)
+                val nextPane = retrievedAuthSession.nextPane
+                if (nextPane == Pane.PARTNER_AUTH) {
+                    // auth session was not completed, proceed with cancellation
+                    cancelAuthSessionAndContinue(authSession = retrievedAuthSession)
+                } else {
+                    // auth session succeeded although client didn't retrieve any deeplink.
+                    navigationManager.navigate(NavigateToRoute(nextPane.toNavigationCommand()))
```

Comment:
may be worthwhile to log exactly what nextPane is here in case we get something unexpected.

Reply:
logged via `auth_session.retrieve` that includes a `next_pane` field.